substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,236 @@
+            Method getReflectionFactoryMethod = ReflectionUtil.lookupMethod(reflectionFactoryClass, "getReflectionFactory");
+            Object reflFactoryInstance = getReflectionFactoryMethod.invoke(null);
+            Method newConstructorForSerializationMethod = ReflectionUtil.lookupMethod(reflectionFactoryClass, "newConstructorForSerialization", Class.class);
+            buildTimeConstructor = (Constructor<?>) newConstructorForSerializationMethod.invoke(reflFactoryInstance, serializationTargetClass);
+        } catch (ReflectiveOperationException e) {
+            throw VMError.shouldNotReachHere(e);
+        }
+        if (buildTimeConstructor == null) {
+            return null;
+        }
+        Class<?> buildTimeConsClass = buildTimeConstructor.getDeclaringClass();
+        // Prepare build time checksum and verify with configured checksum
+        long checksum;
+        // this class is getting from SerializationChecksumClassLoader classloader
+        Class<?> checksumCalculationTargetClass;
+        try {
+            // Checkstyle: stop
+            checksumCalculationTargetClass = Class.forName(serializationTargetClass.getName(), false, serializationChecksumClassLoader);
+            // Checkstyle resume
+        } catch (ClassNotFoundException e) {
+            throw VMError.shouldNotReachHere(e);
+        }
+        long buildTimeChecksum = checksumCalculator.calculateChecksum(buildTimeConsClass.getName(), serializationTargetClass.getName(), checksumCalculationTargetClass);
+        if (configuredChecksum != null) {
+            if (configuredChecksum.longValue() != buildTimeChecksum) {
+                StringBuilder sb = new StringBuilder();
+                sb.append("\nBuild time serialization class checksum verify failure.")
+                                .append(" The classes' hierarchy may have been changed from configuration collecting time to image build time:\n");
+                sb.append(targetClassName).append(": configured checksum is ").append(configuredChecksum).append(", build time checksum is ").append(buildTimeChecksum);
+                reportError((FeatureImpl.BeforeAnalysisAccessImpl) access, CHECKSUM_VERIFY_FAIL, sb.toString());
+            }
+            checksum = configuredChecksum;
+        } else {
+            // If the checksum is not set in configurations, use the build time checksum instead
+            checksum = buildTimeChecksum;
+        }
+
+        // Cache constructorAccessor
+        try {
+            Method getConstructorAccessor = ReflectionUtil.lookupMethod(Constructor.class, "getConstructorAccessor");
+            Object constructorAccessor = getConstructorAccessor.invoke(buildTimeConstructor);
+            CachedEntity exitingEntity = cachedSerializationConstructorAccessors.putIfAbsent(targetClassName,
+                            new CachedEntity(constructorAccessor, checksum));
+            if (exitingEntity != null) {
+                StringBuilder sb = new StringBuilder();
```

Comment:
This is too strict. We have to be able to handle redundant / overlapping `serialization-config.json` entries (jar file from project X might have an embedded serialization config that overlaps with another one in project Y). Otherwise the accumulative nature of our configurations will interact badly with serialization configuration data.

Reply:
Loose this check by adding a checksum compare. Only report error for same name different checksum case.